skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+            self.created_on = instances_seen
+            self.drift_detection.reset()  # TODO check
+        self.evaluator = self.evaluator_method()
+
+    def partial_fit(self, X, y, weight, instances_seen):
+        self.classifier.partial_fit(X, y, weight)
+
+        if self.background_learner:
+            self.background_learner.classifier.partial_fit(X, y, INSTANCE_WEIGHT)
+
+        correctly_classifies = False
+        if self.use_drift_detector and not self.is_background_learner:
+            correctly_classifies = self.classifier.predict(X) == y
+            # Check for warning only if use_background_learner is active
+            if self.use_background_learner:
+                self.warning_detection.add_element(int(not correctly_classifies))
+                # Check if there was a change
+                if self.warning_detection.detected_change():
+                    self.last_warning_on = instances_seen
+                    self.nb_warnings_detected += 1
+                    # Create a new background tree classifier
+                    background_learner = self.classifier.new_instance()
+                    # Create a new background learner object
+                    self.background_learner = ARFBaseLearner(self.index_original,
+                                                             background_learner,
+                                                             instances_seen,
+                                                             self.use_background_learner,
+                                                             self.use_drift_detector,
+                                                             self.evaluator_method,
+                                                             self.drift_detection_method,
+                                                             self.warning_detection_method,
+                                                             True)
+                    # Update the warning detection object for the current object
+                    # (this effectively resets changes made to the object while it was still a bkg learner).
+                    self.warning_detection.reset()  # TODO check
+
+        # Update the drift detection
+        self.drift_detection.add_element(int(not correctly_classifies))
+
+        # Check if there was a change
+        if self.drift_detection.detected_change():
+            self.last_drift_on = instances_seen
+            self.nb_drifts_detected += 1
+            self.reset(instances_seen)
+
+    def predict(self, X):
+        return self.classifier.predict(X)
+    
+    def get_votes_for_instance(self, X):
+        return self.classifier.get_votes_for_instance(X)
+
+    def get_class_type(self):
+        raise NotImplementedError
+
+    def get_info(self):
+        return "NotImplementedError"
+
+
+class ARFBaseClassifierEvaluator(BaseObject):
+    """Basic Classification Performance Evaluator
+    TODO replace with skmultiflow evaluator
```

Comment:
Just to emphasize that it is probably a good thing to use the library evaluator instead of this custom version (it will simplify things a lot).

Reply:
- [x] Done